### PR TITLE
Render image preview of templates on the client side

### PR DESCRIPTION
**Pull request by rosendi**
Please review that commit https://github.com/sendsay-ru/account/commit/e4d1ca6f79c1f5286648a48c081894a149c47671

### CODE REVIEW

**Comment by trikashny** [comment]
How can I rerender previews of previously created templates?

**Comment by trikashny** [comment]
http://snpy.in/KLIQvs There is no background fill, for the short templates, as we discussed.

**Comment by rosendi** [comment]
@trikashny 
> How can I rerender previews of previously created templates? 

You can resave the templates with no background. Is the migration process required?

> There is no background fill, for the short templates, as we discussed.

That's weird, may be `git pull`?

File: src/kit/icons/templatePreview.svg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<svg width="23" height="25" viewBox="0 0 23 25" xmlns="http://www.w3.org/2000/svg">
```
Comment (zouxuoz) [inline src/kit/icons/templatePreview.svg]: size should be 24x24

File: src/resources/file/screenshot/model.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+define(function(require) {
+    var File = require('resources/file/model');
+    var rasterize = require('html2canvas');
+    var app = require('app');
```
Comment (zouxuoz) [inline src/resources/file/screenshot/model.js]: `app` is never used

File: src/kit/icons/templatePreview.svg (hunk 1 above)
Comment (trikashny) [inline src/kit/icons/templatePreview.svg]: `<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 23.998h8.5a.5.5 0 1 0 0-1H1V1h9v7h6.999v4.5a.5.5 0 1 0 1 0V7.295L10.702 0H0v23.998zM16.28 7H11V1.719l5.28 5.28zm3.219 5.046l-8.047 8.093a.5.5 0 0 0-.14.219l-.844 3a.5.5 0 0 0 .61.625l3-.797a.5.5 0 0 0 .218-.14l8.109-8.094c.8-.8.8-2.106 0-2.906-.4-.4-.928-.594-1.453-.594s-1.053.194-1.453.594zm1.875 2.578l-7.672 7.625-2.031.53.563-1.952 7.64-7.703 1.5 1.5zM2 9h14v5H2V9zm0 6h6v1H2v-1zm0 2h4v1H2v-1zm5 0h5v1H7v-1zm-5 2h6v2H2v-2zm12-4v1H9v-1h5zM3 10h12v3H3v-3z" fill="#000"/>
</svg>
`

File: src/resources/file/screenshot/model.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            var renderer = this._constructRenderer();
+
+            renderer.onload = function() {
+                var rendererDocument = self._getRendererDocument(renderer);
+
+                var options = {
+                    background: self._extractMainBackgroundColor(rendererDocument),
+                    letterRendering: true,
+                    useCORS: true,
+                }
```
Comment (zouxuoz) [inline src/resources/file/screenshot/model.js]: missed `semicolon`

**Comment by zouxuoz** [comment]
LGTM

**Pull request merged**